x/erc20/keeper/conversions.go

```diff
@@ -0,0 +1,64 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/evmos/evmos/v12/contracts"
+	"github.com/evmos/evmos/v12/x/erc20/types"
+)
+
+// TryConvertErc20Sdk converts ERC20 token to SDK coin from sender's ETH address to receiver's SDK address.
+// If the sender do not have enough balance, the method returns error without any state changes.
+// The method
+//   - Burns escrowed tokens
+//   - Unescrows coins that have been previously escrowed with ConvertCoin
+func (k Keeper) TryConvertErc20Sdk(
+	ctx sdk.Context,
+	sender, receiver sdk.AccAddress,
+	denom string, // denom may be either ERC20 address or SDK coin
```

Comment:
I guess if `denom` = `0xContractAddress` then `MintingEnabled` will always return error.
Why not only SDK coin?

Reply:
i think it works with both, examples from x/erc20:
- [sdk coin](https://github.com/dymensionxyz/evmos/blob/9748c46cf911e5b0f1e69a88ceb224fffd735c2f/x/erc20/keeper/msg_server.go#L50)
- [erc20 contract](https://github.com/dymensionxyz/evmos/blob/9748c46cf911e5b0f1e69a88ceb224fffd735c2f/x/erc20/keeper/msg_server.go#L92)

but anyway, currently this field always uses sdk coins

Reply:
yes you are right, they work for both